internal/state/graph/graph.go

```diff
@@ -50,7 +51,7 @@ func BuildGraph(
 
 	processedGws := processGateways(state.Gateways, gcName)
 
-	gw := buildGateway(processedGws.Winner, secretMemoryMgr, gc)
+	gw := buildGateway(processedGws.Winner, secretMemoryMgr, gc, state.ReferenceGrants)
```

Comment:
would it be better to pass a type/interface around that incapsulates how reference grants are stored and just provides methods to answer if a reference allowed?

Reply:
Yes, but I was thinking I would add that abstraction when I implement ReferenceGrants from routes to backends so that I can create a better interface.